src/OrbitBase/ThreadPool.cpp

```diff
@@ -179,17 +168,14 @@ void ThreadPoolImpl::WorkerFunction() {
       break;
     }
 
-    bool auto_profile = auto_profile_;
     mutex_.Unlock();
-    if (auto_profile) ORBIT_START("Execute Action");
     action->Execute();
-    if (auto_profile) ORBIT_STOP();
```

Comment:
@pierricgimmig I'm for just removing this, but let me know if you need introspection for `ThreadPool` to be kept. In that case I could move `ThreadPool` out of `OrbitBase` to its own module.

Reply:
I'd prefer keeping it, yes, as it catches a lot of interesting scopes automatically. Could we set some callbacks on the ThreadPool so that we don't have to move it out of OrbitBase? Something like OnActionStarted/OnActionCompleted maybe? I was thinking also that at some point we could provide a name for actions so that they could show up in the introspection scope, that would be a different change.

Reply:
Where or by whom would those callbacks be set?

Reply:
I was thinking wherever ThreadPool::Create is called, we could specify callbacks that would in turn call ORBIT_START/STOP.

Reply:
@dimitry- What do you think? I thought this would complicate the call sites, but it seems that only the `ThreadPool` created in `App.cpp` would be affected.

Reply:
@dimitry- As discussed, I added the possibility to wrap the `Action` in `ScheduleImpl`. Please have a look.